src/workerd/api/tests/js-rpc-test.js

```diff
@@ -286,5 +405,20 @@ export let receiveStubOverRpc = {
     let stub = await env.MyService.makeCounter(17);
     assert.strictEqual(await stub.increment(2), 19);
     assert.strictEqual(await stub.increment(-10), 9);
+
+    // Do multiple concurrent calls, they should be delivered in the order in which they were made.
+    let promise1 = stub.increment(6);
+    let promise2 = stub.increment(4);
+    let promise3 = stub.increment(3);
+    assert.deepEqual(await Promise.all([promise1, promise2, promise3]), [15, 19, 22]);
+  },
+}
+
+export let promisePipelining = {
+  async test(controller, env, ctx) {
+    assert.strictEqual(await env.MyService.makeCounter(12).increment(3), 15);
+
+    assert.strictEqual(await env.MyService.getAnObject(5).foo, 128);
+    assert.strictEqual(await env.MyService.getAnObject(5).counter.increment(7), 12);
   },
```

Comment:
A test verifying the expected behavior if one of these promises is shared across IoContext's and awaited would be good.

Reply:
Although it's not intended to work, there are some cases that do work, and I suppose it's important to make sure they don't break later. Test added:

ce1f3f8607d2468cc7e6eccc3e845329a4f2b7ff